### PR TITLE
[BRPC] Support transfer RowBatch in ProtoBuf Request to Controller Attachment

### LINKED ISSUE #7163: [Bug] Failed to send brpc batch, error_text=Fail to compress request
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

trunk

### What's Wrong?

By default, if the length of ProtoBuf Request> 2G, error will be reported: failed to send brpc batch, error=Bad request, error_text=[E1003]Fail to compress request.

### What You Expected?

1. select
Modify the length of `_brpc_request.row_batch()` in `DataStreamSender::Channel::send_batch`
```
std::string ss = "abcdefghijklmnopqrstuvwxyz_hahahahahaha";
    for (double_t i = 0; i <(25); i++) {// 3G
        ss += ss;
    }
    if (_brpc_request.row_batch().ByteSizeLong() != 0) {
        auto rb = _brpc_request.mutable_row_batch();
        rb->set_tuple_data(ss);
    }
```
error: 
```
Thrift rpc error: failed to send brpc batch, error=Bad request, error_text=[E1003]Fail to compress request, compress_t
pye=0
```

2. load
Modify the length of `_brpc_request.row_batch()` in `NodeChannel::try_send_batch`, error are the same as above.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

Transfer RowBatch in ProtoBuf Request to Controller Attachment, when the maximum length of the RowBatch in the ProtoBuf Request is exceeded.
This can avoid reaching the upper limit of the ProtoBuf Request length (2G), and it is expected that performance can be improved.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #7163) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/proto_util.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+namespace doris {
+
+const int64_t MAX_REQUEST_BYTE_SIZE = 64 * 1024 * 1024;
```
Comment (yangzhg) [inline be/src/util/proto_util.h:22]: Use this as a config instead `brpc_request_transfer_attachment`

File: be/src/util/proto_util.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+inline void row_batch_transfer_attachment(doris::PRowBatch* row_batch, butil::IOBuf* attachment) {
+    row_batch->set_tuple_data_size(row_batch->tuple_data().size());
+    attachment->append(row_batch->tuple_data());
+    row_batch->clear_tuple_data();
+    row_batch->set_tuple_data("");
+}
+
+template <typename Params, typename Closure>
+inline void request_transfer_attachment(Params* brpc_request, Closure* closure) {
```
Comment (yangzhg) [inline be/src/util/proto_util.h:32]: Why not use `row_batch_transfer_attachment()` directly

File: gensrc/proto/data.proto [hunk 3]
```diff
@@ -33,6 +33,7 @@ message PRowBatch {
     repeated int32 tuple_offsets = 3;
     required bytes tuple_data = 4;
     required bool is_compressed = 5;
+    optional int64 tuple_data_size = 6;
```
Comment (yangzhg) [inline gensrc/proto/data.proto:36]: there is no flag to desccribe data is in tuple_data or attachment?

File: be/src/util/proto_util.h (hunk 1 above)
Comment (xinyiZzz) [reply be/src/util/proto_util.h]: done

File: be/src/util/proto_util.h (hunk 2 above)
Comment (xinyiZzz) [reply be/src/util/proto_util.h]: done, combines two methods

File: gensrc/proto/data.proto (hunk 3 above)
Comment (xinyiZzz) [reply gensrc/proto/data.proto]: done

File: gensrc/proto/data.proto [hunk 4]
```diff
@@ -33,6 +33,7 @@ message PRowBatch {
     repeated int32 tuple_offsets = 3;
     required bytes tuple_data = 4;
     required bool is_compressed = 5;
+    optional bool transfer_attachment = 6;
```
Comment (yangzhg) [inline gensrc/proto/data.proto:36]: set default to false

Comment (xinyiZzz) [reply gensrc/proto/data.proto]: done

File: gensrc/proto/data.proto [hunk 5]
```diff
@@ -40,6 +40,7 @@ message PRowBatch {
     repeated int32 tuple_offsets = 3;
     required bytes tuple_data = 4;
     required bool is_compressed = 5;
+    optional bool transfer_attachment = 6 [default = false];
```
Comment (yangzhg) [inline gensrc/proto/data.proto:43]:
```suggestion
    optional bool transfer_by_attachment = 6 [default = false];
```

**Review by yangzhg — DISMISSED** [review]
LGTM

File: be/src/common/config.h [hunk 6]
```diff
@@ -529,6 +529,9 @@ CONF_String(default_rowset_type, "BETA");
 CONF_Int64(brpc_max_body_size, "209715200");
 // Max unwritten bytes in each socket, if the limit is reached, Socket.Write fails with EOVERCROWDED
 CONF_Int64(brpc_socket_max_unwritten_bytes, "67108864");
+// Whether to transfer RowBatch in ProtoBuf Request to Controller Attachment and send it
+// through brpc, this will be faster and avoid the error of Request length overflow.
+CONF_mBool(brpc_request_rowbatch_to_attachment, "false");
```
Comment (yangzhg) [inline be/src/common/config.h:534]:
```suggestion
CONF_mBool(transfer_data_by_brpc_attachment, "false");
```

**Review by yangzhg — COMMENTED** [review]
LGTM

**Review by yangzhg — APPROVED** [review]

File: be/src/common/config.h (hunk 6 above)
Comment (xinyiZzz) [reply be/src/common/config.h]: done

**Pull request merged**